dev/tools/localization/gen_l10n.dart

```diff
@@ -0,0 +1,372 @@
+/// by `@className.of(context)`.
+///
+/// Applications need to include `@className.delegate()` in their app\'s
+/// localizationDelegates list, and the locales they support in the app\'s
+/// supportedLocales list. For example:
+///
+/// ```
+/// import '@importFile';
+///
+/// return MaterialApp(
+///   localizationsDelegates: @className.localizationsDelegates,
+///   supportedLocales: @className.supportedLocales,
+///   home: MyApplicationHome(),
+/// );
+/// ```
+///
+/// ## iOS Applications
+///
+/// iOS applications define key application metadata, including supported
+/// locales, in an Info.plist file that is built into the application bundle.
+/// To configure the locales supported by your app, you’ll need to edit this
+/// file.
+///
+/// First, open your project’s ios/Runner.xcworkspace Xcode workspace file.
+/// Then, in the Project Navigator, open the Info.plist file under the Runner
+/// project’s Runner folder.
+///
+/// Next, select the Information Property List item, select Add Item from the
+/// Editor menu, then select Localizations from the pop-up menu.
+///
+/// Select and expand the newly-created Localizations item then, for each
+/// locale your application supports, add a new item and select the locale
+/// you wish to add from the pop-up menu in the Value field. This list should
+/// be consistent with the languages listed in the @className.supportedLocales
+/// property.
+
+class @className {
+  @className(Locale locale) : _localeName = locale.toString();
+
+  final String _localeName;
+
+  static Future<@className> load(Locale locale) {
+    return initializeMessages(locale.toString())
+      .then<@className>((void _) => @className(locale));
+  }
+  static @className of(BuildContext context) {
+    return Localizations.of<@className>(context, @className);
+  }
+
+  static const LocalizationsDelegate<@className> delegate = _@classNameDelegate();
+
+  /// A list of this localizations delegate along with the default localizations
+  /// delegates.
+  ///
+  /// Returns a list of localizations delegates containing this delegate along with
+  /// GlobalMaterialLocalizations.delegate, GlobalCupertinoLocalizations.delegate,
+  /// and GlobalWidgetsLocalizations.delegate.
+  static const List<LocalizationsDelegate> localizationsDelegates = [
+    delegate,
+    GlobalMaterialLocalizations.delegate,
+    GlobalCupertinoLocalizations.delegate,
+    GlobalWidgetsLocalizations.delegate,
+  ];
+
+  /// A list of this localizations delegate's supported locales.
+  @supportedLocales
+
+@classMethods
+}
+
+class _@classNameDelegate extends LocalizationsDelegate<@className> {
+  const _@classNameDelegate();
+
+  @override
+  Future<@className> load(Locale locale) => @className.load(locale);
+
+  @override
+  bool isSupported(Locale locale) => [@supportedLanguageCodes].contains(locale.languageCode);
+
+  @override
+  bool shouldReload(_@classNameDelegate old) => false;
+}
+''';
+
+const String simpleMethodTemplate = '''
```

Comment:
We could map no-parameter lookups to a getter. Would need a `getterMethodTemplate` .. and some other changes :-).

Reply:
Done!